pkg/sink/codec/debezium/codec.go

```diff
@@ -998,9 +1190,745 @@ func (c *dbzCodec) EncodeValue(
+									} else {
+										jWriter.WriteNullField("comment")
+									}
+									if col.DefaultValue == nil {
+										jWriter.WriteNullField("defaultValueExpression")
+									} else {
+										v, ok := col.DefaultValue.(string)
+										if ok {
+											if strings.ToUpper(v) == "CURRENT_TIMESTAMP" {
+												// https://debezium.io/documentation/reference/3.0/connectors/mysql.html#mysql-temporal-types
+												jWriter.WriteAnyField("defaultValueExpression", "1970-01-01 00:00:00")
+											} else if v == "<nil>" {
+												jWriter.WriteNullField("defaultValueExpression")
+											} else if col.DefaultValueBit != nil {
+												jWriter.WriteStringField("defaultValueExpression", common.UnsafeStringToBinary(v))
+											} else {
+												jWriter.WriteStringField("defaultValueExpression", v)
+											}
+										} else {
+											jWriter.WriteAnyField("defaultValueExpression", col.DefaultValue)
+										}
+									}
+									elems := col.GetElems()
+									if len(elems) != 0 {
+										// Format is ENUM ('e1', 'e2') or SET ('e1', 'e2')
+										jWriter.WriteArrayField("enumValues", func() {
+											for _, ele := range elems {
+												jWriter.WriteStringElement(fmt.Sprintf("'%s'", ele))
+											}
+										})
+									} else {
+										jWriter.WriteNullField("enumValues")
+									}
+
+									jWriter.WriteStringField("typeName", name)
+									jWriter.WriteStringField("typeExpression", expression)
+
+									charsetName := getCharset(col.FieldType)
+									if charsetName != "" {
+										jWriter.WriteStringField("charsetName", charsetName)
+									} else {
+										jWriter.WriteNullField("charsetName")
+									}
+
+									length := getLen(col.FieldType)
+									if length != -1 {
+										jWriter.WriteIntField("length", length)
+									} else {
+										jWriter.WriteNullField("length")
+									}
+
+									scale := getScale(col.FieldType)
+									if scale != -1 {
+										jWriter.WriteFloat64Field("scale", scale)
+									} else {
+										jWriter.WriteNullField("scale")
+									}
+									jWriter.WriteIntField("position", pos+1)
+									jWriter.WriteBoolField("optional", !mysql.HasNotNullFlag(flag))
+
+									autoIncrementFlag := mysql.HasAutoIncrementFlag(flag)
+									jWriter.WriteBoolField("autoIncremented", autoIncrementFlag)
+									jWriter.WriteBoolField("generated", autoIncrementFlag)
+								})
+							}
+						})
+						// if e.TableInfo.Comment != "" {
+						// 	jWriter.WriteStringField("comment", e.TableInfo.Comment)
+						// } else {
+						// }
+						jWriter.WriteNullField("comment")
+
+						// Custom attribute metadata for each table change.
+						// jWriter.WriteArrayField("attributes", func() {
+						// 	jWriter.WriteObjectElement(func() {
+						// 		jWriter.WriteStringField("collation", e.Collate)
+						// 	})
+						// })
+					})
+				})
+			})
+		})
+
+		if !c.config.DebeziumDisableSchema {
```

Comment:
Here seems a lot same code with L1052 and L1787, please extract the same part to a function.